src/ArborX_LinearBVH.hpp

```diff
@@ -197,18 +201,21 @@ class BoundingVolumeHierarchy<
   std::enable_if_t<Kokkos::is_execution_space<FirstArgumentType>::value>
   query(FirstArgumentType const &space, Args &&... args) const
   {
-    BoundingVolumeHierarchy<typename DeviceType::memory_space>::query(
-        space, std::forward<Args>(args)...);
+    base_type::query(space, std::forward<Args>(args)...);
   }
 };
 
+template <typename MemorySpace>
+using BoundingVolumeHierarchy = BasicBoundingVolumeHierarchy<MemorySpace>;
```

Comment:
This is a bit misleading, because template can be both `MemorySpace` and `DeviceType`
```suggestion
template <typename Type>
using BoundingVolumeHierarchy = BasicBoundingVolumeHierarchy<Type>;
```

Reply:
And we do not want people to use `DeviceType` so I explicitly say `MemorySpace`.

Reply:
I guess that's fair. It's better to expose users to the right ideas, even though on the developers side one would need to remember about this.